examples/poisson_example.cpp

```diff
@@ -56,87 +85,77 @@ int main(int argc, char *argv[])
-    gsInfo << "The domain is a "<< patches <<"\n";
-    //! [Geometry data]
-
-    // For single patch unit square of quadratic elements use (Note:
-    // you need to update the bounadry conditions section for this to
-    // work properly!) :
-    // patches = gsMultiPatch<>(*gsNurbsCreator<>::BSplineSquare(2));
+    else if (geo=="BSplineSquareGrid")
+        patches = gsNurbsCreator<>::BSplineSquareGrid(2, 2, 0.5);
 
     // Geometry can also be read from file (if gsMultiPatch):
-    // gsReadFile<>("planar/lshape_p2.xml", patches);
+    else
+    {
+        if (! gsFileManager::fileExists(geo))
+        {
+            gsInfo << "The geometry \"" << geo << "\" is unknown.\nAllowed are only "
+                "\"BSplineQuarterAnnulus\", \"BSplineSquareGrid\" and filenames of "
+                "xml-files containing a gsMultiPatch geometry.\n";
+            return EXIT_FAILURE;
+        }
+        gsReadFile<>(geo, patches);
+    }
+    gsInfo << "Geometry: "<< patches <<"\n";
+    //! [Geometry data]
+
 
+    //! [Boundary conditions]
     // Define Boundary conditions. Note that if one boundary is
     // "free", eg. if no condition is defined, then it is a natural
     // boundary (zero Neumann condition)
-    // Also, remember that a pure Neumann problem has no unique
-    // solution, thereforer implies a singular matrix. In this case
-    // a corner DoF can be fixed to a given value to obtain a unique solution.
-    // (example: bcInfo.addCornerValue(boundary::southwest, value, patch);)
-
-    //! [Boundary conditions]
     gsBoundaryConditions<> bcInfo;
-    // Every patch with a boundary need to be specified. In this
-    // there are in total 8 sides (two for each patch)
-
-    // Dirichlet Boundary conditions
-    // First argument is the patch number
-    bcInfo.addCondition(0, boundary::west,  condition_type::dirichlet, &g);
-    bcInfo.addCondition(1, boundary::west,  condition_type::dirichlet, &g);
-
-    bcInfo.addCondition(1, boundary::north, condition_type::dirichlet, &g);
-    bcInfo.addCondition(3, boundary::north, condition_type::dirichlet, &g);
-
-    // Neumann Boundary conditions
-    gsFunctionExpr<> hEast ("1*pi*cos(pi*1)*sin(pi*2*y)", "3*pi*cos(pi*3)*sin(pi*4*y)",2);
-    gsFunctionExpr<> hSouth("-pi*2*sin(pi*x*1)","-pi*4*sin(pi*x*3)",2);
-
-    bcInfo.addCondition(3, boundary::east,  condition_type::neumann, &hEast);
-    bcInfo.addCondition(2, boundary::east,  condition_type::neumann, &hEast);
-
-    bcInfo.addCondition(0, boundary::south, condition_type::neumann, &hSouth);
-    bcInfo.addCondition(2, boundary::south, condition_type::neumann, &hSouth);
-    //! [Boundary conditions]
-
-    /*
-      //Alternatively: You can automatically create Dirichlet boundary
-      //conditions using one function (the exact solution) for all
-      //boundaries like this:
 
+    // Here, we just define Dirichlet boundary conditions everywhere,
+    // based on the made up solution
```

Comment:
because one can choose the domain and we have a made-up solution, so it is non-trivial to determine the "correct" boundary conditions.

in the following, commented out code and in the *.dox-file, the choice of neumann bc is explained.